apps/sms/views/inbox/js/inbox.js

```diff
@@ -405,7 +405,7 @@ var InboxView = {
       var thread = Threads.get(+id);
 
       var markable = thread && !thread.isDraft &&
-        (isRead || !thread.getDraft());
+        (isRead || !thread.getDraft()) && !(thread.unreadCount ^ isRead);
```

Comment:
nit: I would suggest we split this condition into another variable like `readStatusMatched` and ```
```
var readStatusMatched = !(!!thread.unreadCount ^ isRead);
if (markable && !readStatusMatched) {
...
```
Since we don't call the markThreadRead not because the thread is not markable, but it's not necessary to do so due to the status is already the same, wdyt?
p.s : I use !!thread.unreadCount because I want to make sure the it's boolean, not a number. If the unreadCount could be more than 1(should not happen for real code anyway), this xor checking could be failed.